src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -561,7 +561,8 @@
   [{:keys [server] :as webserver-context} :- ServerContext]
   (when-let [mbean-container (:mbean-container @(:state webserver-context))]
     (log/debug "Cleaning up JMX MBean container")
-    (.destroy mbean-container))
+    (.destroy mbean-container)
+    (swap! (:state webserver-context) assoc :mbean-container nil))
```

Comment:
I'm expecting the work that you're described to be complete when the current TK PRs and TK-331 are finished.  Do you disagree?